templates/vue/src/components/Lightbox/media/MultiContentMedia/PageMenu/MenuGroups.vue

```diff
@@ -0,0 +1,148 @@
+              @handle-menu-item-click="handleMenuItemClick"
+            />
+          </ul>
+        </b-collapse>
+      </b-card>
+    </div>
+  </b-container>
+</template>
+
+<script>
+import { mapGetters } from "vuex"
+import PageMenuItem from "./PageMenuItem"
+export default {
+  name: "menu-groups",
+  components: {
+    PageMenuItem,
+  },
+  props: {
+    node: {
+      type: Object,
+      required: true,
+    },
+    rows: {
+      type: Array,
+      required: true,
+    },
+    disabled: {
+      type: Boolean,
+      required: false,
+      default: false,
+    },
+  },
+  computed: {
+    ...mapGetters(["getParent"]),
+    lockRows() {
+      return this.node.typeData.lockRows
+    },
+    disabledFrom() {
+      return this.rows.findIndex(row => !row.node.completed)
+    },
+    menuGroups() {
+      const menu = []
+      const mainMenu = []
+      this.rows.forEach(row => {
+        row.node.typeData.isSecondaryNode ? menu.push([row]) : mainMenu.push(row)
+      })
+      menu.unshift(mainMenu)
+      return menu
+    },
+    filteredMenuGroups() {
+      const filteredMenu = []
+      this.menuGroups.forEach((menu, mIndex) => {
+        if (mIndex === 0) {
+          filteredMenu.push(menu)
+        } else {
+          let childrenNodes = menu[0].children
+          const newMenu = []
+          childrenNodes.forEach(childNode => {
+            let newRow = {
+              children: [],
+              node: childNode,
+            }
+            newMenu.push(newRow)
+          })
+          filteredMenu.push(newMenu)
+        }
+      })
+      return filteredMenu
+    },
+    menuTitleNodeIds() {
+      let menuTitleNodeIds = []
+      menuTitleNodeIds.push(this.node.id)
+      this.rows.forEach(row => {
+        if (row.node.typeData.isSecondaryNode) {
+          menuTitleNodeIds.push(row.node.id)
+        }
+      })
+      return menuTitleNodeIds
+    },
+  },
+  mounted() {
+    this.handleMenuTitleClick(0)
+  },
+  methods: {
+    disabledRow(node) {
+      const index = this.rows.findIndex(row => row.node.id === node.id)
+      return (
+        (this.lockRows && this.disabledFrom >= 0 && index > this.disabledFrom) ||
+        !node.unlocked
+      )
+    },
+    scrollTo(nodeId) {
+      this.$emit("scroll-to", nodeId)
+    },
+    getMenuName(index) {
+      return index === 0 ? this.node.title : this.menuGroups[index][0].node.title
+    },
+    getMenuIndexFromNodeId(nodeId) {
+      let match = this.menuTitleNodeIds.findIndex(id => id === nodeId)
+      return match > -1 ? match : this.getMenuIndexFromNodeId(this.getParent(nodeId))
+    },
+    getNodeIdFromMenuIndex(index) {
+      return index === 0 ? this.node.id : this.menuGroups[index][0].node.id
+    },
+    handleMenuTitleClick(index) {
+      const pageMenuData = {
+        menuIndex: index,
+        nodeId: this.getNodeIdFromMenuIndex(index),
+        context: index === 0 ? "" : "page",
+      }
+      this.$emit("handle-page-menu-click", pageMenuData)

```

Comment:
Since this is an event, it's better to just call the even something like "title-clicked" or something as the components generally just tell the parent what events are happening and it's up to the parent component to decide to handle that event.